src/libsyntax/ext/quote.rs

```diff
@@ -756,8 +833,23 @@ fn expand_tts(cx: &ExtCtxt, sp: Span, tts: &[ast::TokenTree])
 
     let stmt_let_tt = cx.stmt_let(sp, true, id_ext("tt"), cx.expr_vec_ng(sp));
 
-    let mut vector = vec!(stmt_let_sp, stmt_let_tt);
-    vector.extend(mk_tts(cx, &tts[]).into_iter());
+    vec!(stmt_let_sp, stmt_let_tt)
+}
+
+fn mk_tts(cx: &ExtCtxt, tts: &[ast::TokenTree], matcher: bool) -> Vec<P<ast::Stmt>> {
```

Comment:
`mk_tts` creates statements. It could return a FlatMap instead, but the signature would be complex.